DigitalLearningSolutions.Web/Views/MyAccount/_EditRegistrationNumber.cshtml

```diff
@@ -0,0 +1,63 @@
+﻿@using DigitalLearningSolutions.Data.Enums
+@using DigitalLearningSolutions.Web.ViewModels.MyAccount
+@model EditDetailsViewModel
+
+@{
+  var errorHasOccurred = !ViewData.ModelState.IsValid && @Model.ProfessionalRegNumberSelectionAnswer == YesNoSelectionEnum.None;

```

Comment:
Not sure I like this, it relies too much on how the validation works elsewhere. Could we do something like check specifically where there's an error on the `ProfessionalRegNumberSelectionAnswer` or `ProfessionalRegistrationNumber` properties?

Reply:
I have updated the code. We only show validation in case the professional registration number is selected on postback, when model state is invalid. If we don't add this check, the validation may appear when page loads.